src/methods/SANA.cpp

```diff
@@ -1111,31 +1111,19 @@ void SANA::performChange(int type) {
     double newCurrentScore = 0;
     bool makeChange = scoreComparison(newAligEdges, newInducedEdges, newTCSum, newLocalScoreSum, newWecSum, newNcSum, newCurrentScore, newEwecSum, newSquaredAligEdges, newExposedEdgesNumer);
 
-    if(wasBadMove && !makeChange)
-    {
-        #ifdef CORES
-			// Statistics on the emerging core alignment.
-			// only update pBad if it's nonzero; re-use previous nonzero pBad if the current one is zero.
-			static double pBad;
-			if(trueAcceptingProbability()>0) pBad = trueAcceptingProbability();
-			coreCount[source]++;
-			coreFreq[source][(*A)[source]]++;
-			weightedCoreFreq[source][(*A)[source]] += pBad;
-			totalCoreWeight[source] += pBad;
-		#endif
-    }
+    #ifdef CORES
+		// Statistics on the emerging core alignment.
+		// only update pBad if it's nonzero; re-use previous nonzero pBad if the current one is zero.
+        uint betterHole = wasBadMove ? source : newTargetIndex;
```

Comment:
Good idea, but I think also if you create a "worseHole" variable it'll be useful below. See later comment.